server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -137,7 +137,12 @@ public enum Property {
         /**
          * Indicates a setting that contains sensitive information. Such a setting will be masked when shown to the users.
          */
-        Masked
+        Masked,
+
+        /**
+         * Indicates an index-level setting that is allowed to be changed on a replicated index.
+         */
+        ReplicatedInternalIndex
```

Comment:
The suggested name `ReplicatedInternalIndex` was following the `InternalIndex` which documents: 
```
Indicates an index-level setting that is managed internally. Such a setting can only be added to an index on index creation but can not be updated via the update API.
```

So the opposite of what it is currently used for in your changes.

Reply:
So a setting with the property `ReplicatedInternalIndex` can not be changed on a replicated index ?

Reply:
This was the idea, but the other way around may make more sense, but then the Property should be called different.